bitbucket/request.js

```diff
@@ -139,19 +137,20 @@ module.exports = function Request(_options) {
         path += `?${query}`;
       }
 
-      if (options.use_xhr) {
-        const xhrOptions = {
+      if (options.requesterFn) {
+        const requesterOptions = {
           headers,
-          json: true,
+          hostname,
           method,
-          timeout: options.timeout * 1000,
+          path,
```

Comment:
I think we should provide the querystring-less `path` and the `query` separately.